### PR TITLE
DOC: Tidied up machine learning documentation

### PR DESCRIPTION
**Pull request by tjof2**
As mentioned in #964, I've tidied up the structure and grammar of the Machine Learning documentation page.

### CODE REVIEW

**Comment by tjof2** [comment]
Failing appveyor because of #914

**Comment by dnjohnstone** [comment]
For me the main issue with the machine learning section was  that there is absolutely no reference to the fact that the decomposition() method allows you to use any of the methods from scikit-learn -- for example NMF or SparsePCA. We should be linking to the scikit-learn docs and explaining that these are all available I think....

I can add this in a separate PR if you like, this one serves its purpose of tidying the existing work up somewhat.

**Comment by tjof2** [comment]
@dnjohnstone I agree - on reading/editing it I realised there was a lot missing (as you say, e.g. NMF), but I decided against changing it too much for now. Crucially, the introduction paragraph needs expanding upon:

> HyperSpy provides easy access to several "machine learning" algorithms that can be useful when analysing multi-dimensional data. In particular, decomposition algorithms, such as principal component analysis (PCA), or blind source separation (BSS) algorithms, such as independent component analysis (ICA), are available through the methods described in this section.

Plus a section on NMF (after PCA+denoising, and before or after BSS/ICA) and a link to e.g. the SPED paper showing an example of the use of NMF.

**Comment by dnjohnstone** [comment]
Ok thanks @tjof2.

In my opinion this is moving in the right direction and it should probably fall to someone else i.e. me to make the link to scikit-image and describe other algorithms as I've used them a fair bit. Therefore I would merge this as I can't see any big issues.

**Comment by francisco-dlp** [comment]
It looks really good.

I agree that it should be expanded to include NMF. But then, why renaming the decomposition section to PCA? Wouldn't it be better to have a decomposition section with NMF and PCA as subsections? Also, the poisson noise normalisation applies to all decomposition algorithms, while the scree plot only to PCA. Sparse PCA and fast_svd are also missing.

**Comment by dnjohnstone** [comment]
That's a good point @francisco-dlp I think an overall heading "Decomposition" with "Principal Component Analysis" and "Non-negative matrix factorisation" as sub headings is a good move if only because it mirrors usage...

One thing that has seemed slightly inconsistent to me for some time is the separation of NMF and ICA in the code such that they are implemented in different parent methods. In many ways NMF is much more similar to ICA than to PCA most notably the NMF implementation we use actually first does an SVD to initialise the problem and then makes it non-negative and optimises.

Personally I prefer to think of them all as constrained matrix factorisations with different constraints and would rather have ICA within decomposition and just run the prior PCA first by default... This is way beyond the scope of this PR though!

**Comment by tjof2** [comment]
@francisco-dlp  a valid point, I missed it initially - changed the headings and put scree plots under PCA.

Poisson noise section sits just below PCA, I presume NMF will sit nicely in between,

**Comment by tjof2** [comment]
@francisco-dlp to clarify, is there anything else you'd like me to change in this PR? Certainly there's more to be done to improve the docs on this in future as @dnjohnstone mentions.

**Comment by dnjohnstone** [comment]
To me this is an improvement and so should be merged - we can get to making it even better when we get there I think... although could also leave it as a PR to keep it fresh in people's minds

File: doc/user_guide/mva.rst [hunk 1]
```diff
@@ -4,104 +4,68 @@ Machine learning
...
-estimation of the dimensionality of the data (or equivalently, the number of
-components that should be retained - see below).
+plotting the explained variance against the component index. This plot is
+sometimes called a scree plot and it should drop quickly,
+eventually becoming a slowly descending line.
+
+The point at which the scree plot becomes linear (often referred to as
+the `elbow`) is generally judged to be a good estimation of the dimensionality
+of the data (or equivalently, the number of components that should be retained
+ - see below).
```
Comment (francisco-dlp) [inline doc/user_guide/mva.rst]: an hyphen at the beginning of the line becomes a bullet in rest...

**Comment by francisco-dlp** [comment]
Other than the little typo above, I agree that this is ready to merge.

**Comment by tjof2** [comment]
@francisco-dlp fixed.

**Pull request merged**